language/stdlib/modules/transaction_fee_distribution.mvir

```diff
@@ -0,0 +1,130 @@
+// This module implements a basic transaction fee distribution logic.
+//
+// We have made a couple design decisions here that are worth noting:
+//  * We pay out per-block as opposed to per-epoch. In the current distribution scheme (equal to
+//    all validators) there isn't any real difference between this and per-epoch distribution.
+//    However, other distribution schemes may have different economic implications depending upon
+//    if fees are distributed per-block or per-epoch.
```

Comment:
I thought that we decided to pay out per-epoch to reduce computational and storage costs? cc @sblackshear

Reply:
For some reason I thought we decided on per-block, but now that you mention it I think you're right. However, at the current moment, we don't have that mechanism (and for that matter, we don't even have this plumbed in per-block since that depends upon block metadata transactions). 

As it stands currently, changing it to be epoch-based should only change 3/4ish lines of code, and I would like to stop having to babysit this PR. So how about, for now I'll update the comment and put a TODO in, and then merge this in. Once the block metadata transactions and epoch changes get in I'll then circle back around and add that in?